login/profile_blackbox_test.go

```diff
@@ -0,0 +1,223 @@
+package login_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/oauth2"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/application"
+	"github.com/almighty/almighty-core/auth"
+	config "github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormapplication"
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	"github.com/almighty/almighty-core/gormtestsupport"
+	"github.com/almighty/almighty-core/test"
+	testtoken "github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+
+	"github.com/almighty/almighty-core/login"
+	"github.com/almighty/almighty-core/migration"
+
+	"github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type profileBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	db             *gormapplication.GormDB
+	clean          func()
+	ctx            context.Context
+	profileService login.UserProfileService
+	configuration  *config.ConfigurationData
+	loginService   *login.KeycloakOAuthProvider
+}
+
+func TestRunProfileBlackBoxTest(t *testing.T) {
+	suite.Run(t, &profileBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
+}
+
+// SetupSuite overrides the DBTestSuite's function but calls it before doing anything else
+// The SetupSuite method will run before the tests in the suite are run.
+// It sets up a database connection for all the tests in this suite without polluting global space.
+func (s *profileBlackBoxTest) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+
+	// Make sure the database is populated with the correct types (e.g. bug etc.)
+	if _, c := os.LookupEnv(resource.Database); c != false {
+		if err := models.Transactional(s.DB, func(tx *gorm.DB) error {
+			s.ctx = migration.NewMigrationContext(context.Background())
+			return migration.PopulateCommonTypes(s.ctx, tx, workitem.NewWorkItemTypeRepository(tx))
```

Comment:
> But do you actually need the DB here? It's not used in you profile service, is it?
If not, then you can convert this test to remote test.

Correct. Ive converted into a remote test.
Added a `gormtestsupport/remote_test_suite.go` which has the db stuff stripped out, and used that.

https://github.com/almighty/almighty-core/pull/1035/commits/249a02d7f26270889c22e910456ef92b234624c9